### ISSUE TITLE
Give @TianyuZuo root access to test boxes to install docker

### ISSUE THREAD

**Issue by gdams** [AdoptOpenJDK/openjdk-infrastructure#29]
@TianyuZuo has offered to install docker on the test machines to aid the docker based testing. I will add their key to the root users for all test machines and will revoke access upon completion

**Comment by TianyuZuo** [comment]
Thanks for your great help George, I will update the progress here.

**Comment by karianna** [comment]
Is this something that could be added to ansible scripts?

**Comment by TianyuZuo** [comment]
Hi @karianna 

It would be great to use ansible scripts to set up docker environment,  could you help to point me which files need the update?

This docker environment is required to run dockerized tests. So they are required on test machine. I am not sure if build machine also need this docker environment. Will the modification on ansible scripts also affect build machines?

**Comment by TianyuZuo** [comment]
two linux ubuntu machines have been installed docker, trying to figure out which docker to install on Mac machines. By using homebrew, it has two dockers in `brew` and `brew cask`, need to find out which is the one we need.

**Comment by karianna** [comment]
Paging @bblondin to assist with the questions on Ansible

The ansible scripts / playbooks are in this infrastructure repo.  I'm not 100% sure how they're split between build and test hosts.

**Comment by gdams** [comment]
so we currently have two sections in each playbook..... (build and test) e.g https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/ubuntu.yml#L28-L63

**Comment by gdams** [comment]
@TianyuZuo the newer docker for mac (which is what we almost certainly want) is installed using `brew cask install docker`

**Comment by TianyuZuo** [comment]
Hi @gdams 
The Mac machines seem have permission issue to instal Docker for Mac

```
DXU751:~ admin$ brew cask install docker
==> Satisfying dependencies
==> Downloading https://download.docker.com/mac/stable/18433/Docker.dmg
Error: Download failed on Cask 'docker' with message: Permission denied - /usr/local/var/homebrew/locks/docker--17.06.0-ce-mac18,18433.dmg.incomplete.lock
Error: Install incomplete.

DXU751:~ admin$ brew update
Error: /usr/local/Cellar is not writable. You should change the
ownership and permissions of /usr/local/Cellar back to your
```

As the docker environment has been installed in two linux machines, I think we can start the container tests on linux first. One minor and silly question is that: how will the ansible script update the machines? If I have a change set merged into the main repo, will all the machines be updated instantly or be updated at the next machine maintenance day?

**Comment by gdams** [comment]
@TianyuZuo have you tried running with sudo? that shouldn't be happening

**Comment by TianyuZuo** [comment]
sudo requires user password which I don't think I have 😞

**Comment by gdams** [comment]
really? That user account I gave you shouldn't need the password for root, let me take a look at that

**Comment by TianyuZuo** [comment]
```
terrys-mbp:jck terryzuo$ ssh admin@74.80.250.151
Last login: Wed Oct 25 13:32:27 2017 from 32.97.110.55
DXU751:~ admin$ sudo brew update
Password:
DXU751:~ admin$ 
DXU751:~ admin$ sudo brew cask install docker
Password:

```

I think Mac has another layer of security check, even the current user is root or under root group. I tried the same command on my Mac (I am root user) and it ask me to re-enter the pssw as well.

**Comment by gdams** [comment]
Yeah the sudoers file needed to be updated. You should now be able to run sudo commands without the root password, let me know if you have any further issues.

**Comment by karianna** [comment]
@TianyuZuo Can this now be closed?

**Comment by TianyuZuo** [comment]
yes please, the work is done. Thanks a lot for the help 👍  @gdams  and @karianna

**Comment by gdams** [comment]
Keybox access is now revoked
